### PR TITLE
added a file_name and directory argument. refactored the search to open in less.

### PR DESCRIPTION
**Pull request by smcalilly**
Hi, I appreciate the simplicity of your script, so hopefully I didn't make it too complicated. I like the utility of it and couldn't help but add a feature. My PR allows you to pass a CLI argument with a file name or a directory within the notes directory. I also made the `grep` search output into `less`. I don't know bash very well, so if you like this idea and are willing to accept a PR with this feature, please let me know ways I can improve the code. Thanks !

### CODE REVIEW

File: src/note.sh [hunk 1]
```diff
@@ -11,20 +11,47 @@ function usage() {
...
 
+# parse any arguments
+while getopts g:d:f: flag 
+do
+    case "${flag}" in
+        f) file_name=${OPTARG};;
+        d) directory=${OPTARG};;
+        g) grp=${OPTARG};;
+    esac
+done
```
Comment (hachibu) [inline src/note.sh:27]: Very cool way to parse the flags that I didn't know about!

File: src/note.sh [hunk 2]
```diff
@@ -11,20 +11,47 @@ function usage() {
...
+    else
+        NOTE_PATH="$NOTE_DIR/${file_name}.md"
+    fi
+  $EDITOR $NOTE_PATH
+
+# search with grep and open the files with less
+elif [ -n "${grp+1}" ]; then
+    notes=$(grep -nr "$2" $NOTE_DIR -l)
+    echo $notes
+    less -p "$2" $notes
```
Comment (hachibu) [inline src/note.sh:54]: I like this idea of using less, maybe I'll try it out.

**Comment by hachibu** [comment]
Thanks for submitting a PR. I learned some new Bash things from reading your code.

I'm not quite sure where I want to take the project so I can't accept this at the moment. But I'm going to experiment with less a little bit.

Again, thanks for opening a PR!

**Comment by obar1** [comment]
np just make a new branch with some pieces you like and go ahead ;) @hachibu

**Pull request closed without merging**